### DISCUSSION TITLE
How to create a custom font?

### DISCUSSION THREAD

**Discussion by TESSON-MAKER** [olikraus/u8g2#1508] [General]
Hello everyone, my name is Paul and I am just new to Arduino programming. For a personal project, my goal is to display the time on my 128 * 64 and I would like to create my own font. How to do ?
During this time, I have already looked at the file listing the different fonts and I don't know how to use it (u8g2_fonts), can you help me?

(I saw the previous discussion about fonts and I'm lost)

**Comment by olikraus** [comment]
> create my own font. How to do ?

Did you check the FAQ? There are several questions starting from here: https://github.com/olikraus/u8g2/blob/c7dc077995ca513337a7d2a1970df693e18ee308/doc/faq.txt#L245

Background: U8g2 requires a font in .bdf format. From there the font is converted into a .c file. 
The code in .c can be just pasted into your .ino script and the array name assigned to u8g2.setFont().
There are also tools to create .bdf from a .ttf font (fontforge, otf2bdf), but maybe you should specify more cleary what you already have.

> know how to use it (u8g2_fonts)

Just call setFont() with the font name. Arduino will take care on the rest.

**Reply by TESSON-MAKER** [reply]
Sorry ! Thanks you !

**Reply by olikraus** [reply]
no problem, let me know if you have further questions.
